src/main/java/com/cryptomorin/xseries/skull/XSkull.java

```diff
@@ -222,53 +222,115 @@ public final class XSkull {
+     * @return The skin value as a {@link String}, or {@code null} if not found.
+     * @throws NullPointerException if {@code meta} is {@code null}.
+     */
     @Nullable
     public static String getSkinValue(@Nonnull ItemMeta meta) {
         Objects.requireNonNull(meta, "Skull meta cannot be null");
         GameProfile profile = getProfile(meta);
         return profile == null ? null : getSkinValue(profile);
     }
 
+    /**
+     * Retrieves the skin value from the given {@link BlockState}.
+     *
+     * @param state The {@link BlockState} to retrieve the skin value from.
+     * @return The skin value as a {@link String}, or {@code null} if not found.
+     * @throws NullPointerException if {@code state} is {@code null}.
+     */
     @Nullable
     public static String getSkinValue(@Nonnull BlockState state) {
         Objects.requireNonNull(state, "Block state cannot be null");
         GameProfile profile = getProfile(state);
         return profile == null ? null : getSkinValue(profile);
     }
 
+    /**
+     * Retrieves the skin value from the given {@link GameProfile}.
+     *
+     * @param profile The {@link GameProfile} to retrieve the skin value from.
+     * @return The skin value as a {@link String}, or {@code null} if not found.
+     * @throws NullPointerException if {@code profile} is {@code null}.
+     */
     @Nullable
     public static String getSkinValue(@Nonnull GameProfile profile) {
+        Objects.requireNonNull(profile, "Game profile cannot be null");
         return Optional.ofNullable(Iterables.getFirst(profile.getProperties().get("textures"), null))
                 .map(XSkull::getPropertyValue).orElse(null);
     }
 
+    /**
+     * Retrieves a {@link GameProfile} based on the provided input string.
+     *
+     * @param input The input string to retrieve the profile for.
+     * @return The {@link GameProfile} corresponding to the input.
+     * @throws NullPointerException if {@code input} is {@code null}.
+     */
+    @Nonnull
+    public GameProfile getProfile(@Nonnull String input) {
+        Objects.requireNonNull(input, "Input cannot be null");
+        return getProfile(InputType.get(input), input);
+    }
+
+    /**
+     * Retrieves a {@link GameProfile} based on the provided input type and input string.
+     *
+     * @param type The type of the input.
+     * @param input The input string to retrieve the profile for.
+     * @return The {@link GameProfile} corresponding to the input type and input string.
+     *         Returns a profile with an invalid skull value if the type is {@code null}.
+     * @throws NullPointerException if {@code input} is {@code null}.
+     * @throws AssertionError if the input type is unknown.
+     */
     @Nonnull
     public static GameProfile getProfile(@Nullable InputType type, @Nonnull String input) {
+        Objects.requireNonNull(input, "Input cannot be null");
         if (type == null) return profileFromBase64(INVALID_SKULL_VALUE);
```

Comment:
You should make all the profileFrom...() methods as a abstract method of InputType enum instead. That way this entire method would be useless as well.